lib/access_control.js

```
@@ -0,0 +1,64 @@
+const {
+  useAccessControl
+} = require('../config/email_constants');
+
+
+const requestAuthReduce = (request) => {
+  const isAuthenticated = !!request.auth.credentials?.id;
+  const isAdmin = isAuthenticated && request.auth.credentials.scope.includes('admin');
+  const wantsHidden = request.query.hidden === true;
+  return [isAuthenticated, isAdmin, wantsHidden];
+};
+
+const initializeQuery = (request, entityType) => {
+  const [isAuthenticated, isAdmin, wantsHidden] = requestAuthReduce(request);
+  const hiddenField = `${entityType}_hidden`;
+  const accessListField = `${entityType}_access_list`;
+  const query = { [hiddenField]: false };
+
+  // Admins always get what they want.
+  if (isAdmin) {
+    // Do not specify in query in order to return everything
+    if (wantsHidden) {
+      delete query[hiddenField];
+    }
+
+    return query;
+  }
+
+  // Should not be asking for hidden otherwise.
+  if (wantsHidden) {
+    return undefined;
```

I would just set `wantsHidden = false` and continue along. Returning an undefined query fragment pushes more error checking on the caller. I don't think it's important to try to return an error about this, rather than just silently ignore.